app/routes/administrative-units/administrative-unit/sites/site/index.js

```diff
@@ -0,0 +1,33 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class AdministrativeUnitsAdministrativeUnitSitesSiteIndexRoute extends Route {
+  @service store;
+
+  async model() {
+    let { id: administrativeUnitId } = this.paramsFor(
+      'administrative-units.administrative-unit'
+    );
+
+    let administrativeUnit = await this.store.findRecord(
+      'administrative-unit',
+      administrativeUnitId,
+      {
+        reload: true,
+        include: [
+          'primary-site.address',
+          'primary-site.contacts',
+          'sites.address',
+          'sites.contacts',
+        ].join(),
+      }
+    );
+
+    return {
+      administrativeUnit: administrativeUnit,
+      primarySite: await administrativeUnit.primarySite,
```

Comment:
I think this can be replaced with a simple boolean `isPrimarySite` and do the comparison that you do in the template here instead. The primarySite record isn't needed in the template for anything else I think.

Reply:
I solved it in a different way, by accessing to @model.administrativeUnit.primarySite in the template (we should include administrativeUnit in any cas fot the nale as discussed above).